.github/workflows/macos.yml

```diff
@@ -1,8 +1,18 @@
 name: macos
 
 on:
-  push:
+  workflow_dispatch:
```

Comment:
What's the reason?

Reply:
The `macos` workflow run twice for each commit pushed.
The change allows running the workflow manually using a specific branch and preserving the behavior for the `main` branch.